app/forms/personal_key_form.rb

```diff
@@ -21,8 +21,8 @@ def submit
   attr_reader :user, :success
 
   def valid_personal_key?
-    word_regexp = /\w{#{RandomPhrase::WORD_LENGTH}}/
-    return false unless code =~ /\A#{word_regexp} #{word_regexp} #{word_regexp} #{word_regexp}\Z/
+    length = RandomPhrase::WORD_LENGTH
+    return false unless code =~ /^(?:[a-zA-Z0-9]{#{length}}([\s-])?){3}[a-zA-Z0-9]{#{length}}$/
```

Comment:
I like how the previous regex used `#{word_regexp}`, maybe we could do...

```ruby
valid_characters = "[a-zA-Z0-9]"
recovery_code_length = Figaro.env.recovery_code_length.to_i
return false unless code =~ /(#{valid_characters}{#{length}}([\s-])?){#{recovery_code_length - 1}}#{valid_characters}{#{length}}/
```

```
Also:
- What if we pulled the number of words from our config env var for consistency?
- It's very important to anchor with `\A...\Z` instead of `^...$`